### ISSUE TITLE
[Bug] Uncaught TypeError: Cannot destructure property `catalogItems` of 'undefined' or 'null

### ISSUE THREAD

**Issue by jeffcorpuz** [reactioncommerce/reaction#4736]
## Issue Description
After running fresh installation steps, several console errors appear concerning `catalogItems`.

```
Uncaught TypeError: Cannot destructure property `catalogItems` of 'undefined' or 'null'.
    at React.createElement (app.js:46995)
    at Query.render (modules.js:244359)
    at finishClassComponent (modules.js:188388)
    at updateClassComponent (modules.js:188350)
    at beginWork (modules.js:189019)
    at performUnitOfWork (modules.js:191058)
    at workLoop (modules.js:191097)
    at HTMLUnknownElement.callCallback (modules.js:175294)
    at Object.invokeGuardedCallbackDev (modules.js:175332)
    at invokeGuardedCallback (modules.js:175381)
```

```
The above error occurred in the <Query> component:
    in Query (created by CatalogItems)
    in CatalogItems (created by Tag)
    in Tag (created by Query)
    in Query (created by ShopId)
    in ShopId (created by Tracker(ShopId))
    in Tracker(ShopId) (created by Reaction(Products))
    in Reaction(Products) (created by Tracker(Reaction(Products)))
    in Tracker(Reaction(Products)) (created by Reaction(coreLayout))
    in main (created by Reaction(coreLayout))
    in div (created by Reaction(coreLayout))
    in Reaction(coreLayout) (created by Route)
    in Route
    in Switch (created by Reaction(App))
    in div (created by Reaction(App))
    in Reaction(App) (created by Tracker(Reaction(App)))
    in Tracker(Reaction(App))
    in TranslationProvider (created by Tracker(TranslationProvider))
    in Tracker(TranslationProvider)
    in Router (created by BrowserRouter)
    in BrowserRouter
    in ApolloProvider

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
logCapturedError @ modules.js:189421
```

```
modules.js:248816 Uncaught TypeError: Cannot destructure property `catalogItems` of 'undefined' or 'null'.
    at React.createElement (app.js:46995)
    at Query.render (modules.js:244359)
    at finishClassComponent (modules.js:188388)
    at updateClassComponent (modules.js:188350)
    at beginWork (modules.js:189019)
    at performUnitOfWork (modules.js:191058)
    at workLoop (modules.js:191097)
    at renderRoot (modules.js:191137)
    at performWorkOnRoot (modules.js:191755)
    at performWork (modules.js:191677)
```

## Steps to Reproduce
1. Have a fresh git install of `reaction` with a fresh DB
2. Run `docker-compose up` 
3. Follow logs after successful build, appears in JS console in browser and/or terminal 

### Versions
Reaction Version: 2.0.0-rc.5
Node : 10.10.0
Chrome 69.0.3497.100 / Mac OSX

**Comment by zenweasel** [comment]
I see this issue as well on a fresh install. Getting this error in the server console:

```javascript
06:59:54.641Z ERROR Reaction:  (errorId=cjn9y85n50000b0wbtagik846, userId=9HcPvM3SZugvBwZod)
  path: [
    "catalogItems",
    "edges",
    0,
    "node",
    "product",
    "pricing",
    0,
    "currencyExchangePricing",
    "minPrice"
  ]
Error: Cannot return null for non-nullable field CurrencyExchangeProductPricingInfo.minPrice.
```

**Comment by kieckhafer** [comment]
I was seeing this error earlier. I'm not sure the cause. It seems like #4731 will fix this permanently, but temporarily, since I was unable to see any UI, I fixed this by going into Robo3T and manually adding the Open Exchange Rates `appId` to `settings/openexchangerates/appid` in the `core` package in the `Packages` collection.

<img width="1114" alt="robo_3t_-_1_1" src="https://user-images.githubusercontent.com/4482263/46979649-fa663f00-d086-11e8-91ca-726b1150d601.png">

**Comment by spencern** [comment]
@jeffcorpuz did this get resolved by #4731?

**Comment by jeffcorpuz** [comment]
@spencern yes, it has been resolved if you use `reaction-platform` to run the application. 

However if you run it `docker-compose up` under the `reaction` directory it will run into the mongo init replica issue : https://github.com/reactioncommerce/reaction/issues/4745

Will close and continue testing through `reaction-platform`

____

Resolved By : #4731

**Comment by spencern** [comment]
@jeffcorpuz thanks. I do believe that we're resolving that mongo init issue as well in #4748
